File: backend/src/main/java/net/tape/api/SearchHit.java
package net.tape.api;

/** A single unified search result. {@code url} is the site-relative path for the client. */
public record SearchHit(
    String type,      // show | episode | video | article
    String slug,
    String title,
    String subtitle,
    String imageUrl,
    String url) {}

File: web/src/api/components/schemas/SearchHit.ts
import { Option } from 'fp-ts/lib/Option';
import { string, type } from 'io-ts';
import { optionFromNullable } from 'io-ts-types/lib/optionFromNullable';

export type SearchHit = {
	type: Option<string>;
	slug: Option<string>;
	title: Option<string>;
	subtitle: Option<string>;
	imageUrl: Option<string>;
	url: Option<string>;
};
export const SearchHitIO = type(
	{
		type: optionFromNullable(string),
		slug: optionFromNullable(string),
		title: optionFromNullable(string),
		subtitle: optionFromNullable(string),
		imageUrl: optionFromNullable(string),
		url: optionFromNullable(string),
	},
	'SearchHit',
);
